pkg/storage/odf.go

```diff
@@ -0,0 +1,297 @@
+	// storageSystem object is created.
+	errorMsg string
+}
+
+// NewStorageSystemBuilder creates a new instance of Builder.
+func NewStorageSystemBuilder(apiClient *clients.Settings, name, nsname string) *StorageSystemBuilder {
+	glog.V(100).Infof(
+		"Initializing new storageSystem structure with the following params: %s, %s", name, nsname)
+
+	if apiClient == nil {
+		glog.V(100).Infof("storageSystem 'apiClient' cannot be empty")
+
+		return nil
+	}
+
+	builder := &StorageSystemBuilder{
+		apiClient: apiClient.Client,
+		Definition: &odfoperatorv1alpha1.StorageSystem{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      name,
+				Namespace: nsname,
+			},
+		},
+	}
+
+	if name == "" {
+		glog.V(100).Infof("The name of the storageSystem is empty")
+
+		builder.errorMsg = "storageSystem 'name' cannot be empty"
+
+		return builder
+	}
+
+	if nsname == "" {
+		glog.V(100).Infof("The namespace of the storageSystem is empty")
+
+		builder.errorMsg = "storageSystem 'nsname' cannot be empty"
+
+		return builder
+	}
+
+	return builder
+}
+
+// PullStorageSystem gets an existing storageSystem object from the cluster.
+func PullStorageSystem(apiClient *clients.Settings, name, namespace string) (*StorageSystemBuilder, error) {
+	glog.V(100).Infof("Pulling existing storageSystem object %s from namespace %s",
+		name, namespace)
+
+	if apiClient == nil {
+		glog.V(100).Infof("The apiClient is empty")
+
+		return nil, fmt.Errorf("storageSystem 'apiClient' cannot be empty")
+	}
+
+	builder := StorageSystemBuilder{
+		apiClient: apiClient.Client,
+		Definition: &odfoperatorv1alpha1.StorageSystem{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      name,
+				Namespace: namespace,
+			},
+		},
+	}
+
+	if name == "" {
+		glog.V(100).Infof("The name of the storageSystem is empty")
+
+		return nil, fmt.Errorf("storageSystem 'name' cannot be empty")
+	}
+
+	if namespace == "" {
+		glog.V(100).Infof("The namespace of the storageSystem is empty")
+
+		return nil, fmt.Errorf("storageSystem 'namespace' cannot be empty")
+	}
+
+	if !builder.Exists() {
+		return nil, fmt.Errorf("storageSystem object %s does not exist in namespace %s",
+			name, namespace)
+	}
+
+	builder.Definition = builder.Object
+
+	return &builder, nil
+}
+
+// Get fetches existing storageSystem from cluster.
+func (builder *StorageSystemBuilder) Get() (*odfoperatorv1alpha1.StorageSystem, error) {
+	if valid, err := builder.validate(); !valid {
+		return nil, err
+	}
+
+	glog.V(100).Infof("Getting existing storageSystem with name %s from the namespace %s",
+		builder.Definition.Name, builder.Definition.Namespace)
+
+	storageSystemObj := &odfoperatorv1alpha1.StorageSystem{}
+	err := builder.apiClient.Get(context.TODO(), goclient.ObjectKey{
+		Name:      builder.Definition.Name,
+		Namespace: builder.Definition.Namespace,
+	}, storageSystemObj)
+
+	if err != nil {
+		glog.V(100).Infof("storageSystem object %s does not exist in namespace %s",
```

Comment:
This isn't necessarily an error due to the object not existing. I would just change the glog message to indicate that an error occurred

Reply:
changed, but I'm not so sure it is what you meant.